File: go/src/algorithms/mixed/array-change/change.go
package mixed

func arrayChange(input []int) int {
	count := 0
	for i := 1; i < len(input); i++ {
		if input[i] <= input[i-1] {
			difference := input[i-1] - input[i]
			count += difference + 1
			input[i] = input[i-1] + 1
		}
	}
	return count
}

File: ts/src/algorithms/mixed/array-change/change.ts
export function arrayChange(inputArray: number[]): number {
  let count = 0;
  for (let i = 1; i < inputArray.length; i++) {
    if (inputArray[i] <= inputArray[i - 1]) {
      let difference = inputArray[i - 1] - inputArray[i];
      count += difference + 1;
      inputArray[i] = inputArray[i - 1] + 1;
    }
  }
  return count;
}
